solr/solrj/src/java/org/apache/solr/client/solrj/impl/SolrClientNodeStateProvider.java

```diff
@@ -398,6 +337,39 @@ public Map getZkJson(String path) throws KeeperException, InterruptedException {
       return Utils.getJson(zkClientClusterStateProvider.getZkStateReader().getZkClient(), path, true);
     }
 
+    /**
+     * Will attempt to call {@link #invoke(String, String, SolrParams)} up to five times, retrying on any IO Exceptions
+     */
+    public SimpleSolrResponse invokeWithRetry(String solrNode, String path, SolrParams params) throws InterruptedException, IOException, SolrServerException {
+      int retries = 5;
+      int cnt = 0;
+
+      while (cnt++ < retries) {
+        try {
+          return invoke(solrNode, path, params);
+        } catch (SolrException | SolrServerException | IOException e) {
+          boolean hasIOExceptionCause = false;
+
+          Throwable t = e;
+          while (t != null) {
+            if (t instanceof IOException) {
+              hasIOExceptionCause = true;
+              break;
+            }
+            t = t.getCause();
+          }
+
+          if (hasIOExceptionCause) {
+            log.info("Error on getting remote info, trying again: " + e.getMessage());
```

Comment:
I recommend {} and passing "e".  (note Erick's efforts lately to tame the loggers)